geode-core/src/distributedTest/java/org/apache/geode/internal/cache/PartitionedRegionClearDUnitTest.java

```diff
@@ -171,38 +183,172 @@ private void verifyCacheListenerTriggerCount(MemberVM serverVM) {
     }
   }
 
+  SerializableCallableIF<Integer> getWriterClears = () -> {
+    int clears =
+        clearsByRegion.get(REGION_NAME) == null ? 0 : clearsByRegion.get(REGION_NAME).get();
+    return clears;
+  };
+
+  SerializableCallableIF<Integer> getWriterDestroys = () -> {
+    int destroys =
+        destroyesByRegion.get(REGION_NAME) == null ? 0 : destroyesByRegion.get(REGION_NAME).get();
+    return destroys;
+  };
+
+  void configureServers(boolean dataStoreWithWriter, boolean accessorWithWriter) {
+    dataStore1.invoke(() -> initDataStore(true, dataStoreWithWriter));
+    dataStore2.invoke(() -> initDataStore(true, dataStoreWithWriter));
+    dataStore3.invoke(() -> initDataStore(true, dataStoreWithWriter));
+    accessor.invoke(() -> initAccessor(true, accessorWithWriter));
+    // make sure only datastore3 has cacheWriter
+    dataStore1.invoke(() -> {
+      Region region = getRegion(false);
+      region.getAttributesMutator().setCacheWriter(null);
+    });
+    dataStore2.invoke(() -> {
+      Region region = getRegion(false);
+      region.getAttributesMutator().setCacheWriter(null);
+    });
```

Comment:
It seems inefficient and confusing to set the cache writer for all three data stores, then to immediately remove the writers for datastores1 and 2. Could you instead just pass `false` as the second argument for `initDataStore()` for those datastores?

Reply:
I don't have to remove them. But I want to make sure the test is more deterministic.

Reply:
I agree with Donal, doing this we are testing the behavior of AttributesMutator, not for clear.